### PR TITLE
Convert Groovy views to Jelly | use Jenkins.get()

### PR DESCRIPTION
**Pull request by Wadeck**
- convert groovy view to jelly
- use Jenkins.get()
- ~reformat code to follow Java convention~
- ~reorder pom file parts to follow [convention](https://maven.apache.org/developers/conventions/code.html)~
- ~rename test variables to follow Jenkins standard~

@cloudbees/team-foundation @reviewbybees

### CODE REVIEW

File: src/main/resources/hudson/tasks/AntWrapper/config.jelly [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+The MIT License
+
+Copyright (c) 2016, CloudBees, Inc.
```
Comment (jvz) [inline src/main/resources/hudson/tasks/AntWrapper/config.jelly]:
```suggestion
Copyright (c) 2018, CloudBees, Inc.
```

Comment (batmat) [inline src/main/resources/hudson/tasks/AntWrapper/config.jelly]:
```suggestion
Copyright (c) 2016-2018, CloudBees, Inc.
```
maybe?

File: src/main/resources/hudson/tasks/_ant/AntTargetNote/DescriptorImpl/outline.jelly [hunk 2]
```diff
@@ -21,17 +21,16 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
...
+        <table class='pane' id='console-outline'>
+            <tr>
+                <td class='pane-header'>${%Executed Ant Targets}</td>
+            </tr>
+            <tr>
+                <td id='console-outline-body'/>
+            </tr>
+        </table>
+    </l:ajax>
+</j:jelly>
```
Comment (batmat) [inline src/main/resources/hudson/tasks/_ant/AntTargetNote/DescriptorImpl/outline.jelly]: /me not convinced about this kind of change: IMO though tempting, we should refrein from reformating unrelated code to make code archeology easier.

File: pom.xml [hunk 3]
```diff
@@ -1,105 +1,146 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-  <modelVersion>4.0.0</modelVersion>
-  <parent>
-    <groupId>org.jenkins-ci.plugins</groupId>
-    <artifactId>plugin</artifactId>
-    <version>2.31</version>
-    <relativePath />
-  </parent>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
```
Comment (jglick) [inline pom.xml]: Please do not reformat existing lines of code.

File: src/main/java/hudson/tasks/Ant.java [hunk 4]
```diff
@@ -1,18 +1,18 @@
 /*
  * The MIT License
- * 
+ *
```
Comment (jglick) [inline src/main/java/hudson/tasks/Ant.java]: ditto

**Comment by Wadeck** [comment]
I rebase the PR after the merge of #32. Also remove some of the pure tab/space reformat to keep mostly the relevant one that increase readability.

File: src/test/java/hudson/tasks/_ant/AntTargetAnnotationTest.java [hunk 5]
```diff
@@ -23,7 +23,8 @@
 public class AntTargetAnnotationTest {
     
     @Rule
-    public JenkinsRule r = new JenkinsRule();
```
Comment (jglick) [inline src/test/java/hudson/tasks/_ant/AntTargetAnnotationTest.java]: I have actually been preferring the name `r`, since `j` is too easily confused with the `Jenkins` instance itself.

File: src/test/java/hudson/tasks/_ant/AntTargetNoteTest.java [hunk 6]
```diff
@@ -5,7 +5,7 @@
 import org.junit.Before;
 import org.junit.Test;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
```
Comment (jglick) [inline src/test/java/hudson/tasks/_ant/AntTargetNoteTest.java]: Common style for static imports in tests is to use wildcards.

File: src/main/java/hudson/tasks/Ant.java [hunk 7]
```diff
@@ -104,7 +104,7 @@
     private final String properties;
     
     @DataBoundConstructor
-    public Ant(String targets,String antName, String antOpts, String buildFile, String properties) {
+    public Ant(String targets, String antName, String antOpts, String buildFile, String properties) {
```
Comment (jglick) [inline src/main/java/hudson/tasks/Ant.java]: Unfortunately https://github.com/jenkinsci/ant-plugin/pull/33/files?w still does not ignore this, making this diff too long to be read.

**Comment by jglick** [comment]
Anyway I am done “maintaining” this—just needed to deliver #32 urgently.

File: src/test/java/hudson/tasks/_ant/AntTargetAnnotationTest.java (hunk 5 above)
Comment (Wadeck) [reply src/test/java/hudson/tasks/_ant/AntTargetAnnotationTest.java]: My concern here was essentially because the same name is used for `RestartableJenkinsRule r`. No worries I revert them.

Comment (jglick) [reply src/test/java/hudson/tasks/_ant/AntTargetAnnotationTest.java]: My style (FWIW!) is to use `RestartableJenkinsRule rr`.

File: src/main/resources/hudson/tasks/Ant/config.jelly [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
+    <j:if test="${size(descriptor.installations) != 0}">
```
Comment (jglick) [inline src/main/resources/hudson/tasks/Ant/config.jelly]: You tested this logic interactively, right? (Cannot recall if there is a functional test for it.)

File: src/main/resources/hudson/tasks/Ant/config.jelly [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        <f:entry title="${%Ant Version}">
+            <select class="setting-input" name="ant.antName">
+                <option value="(Default)">${%Default}</option>
+                <j:forEach var="install" items="${descriptor.installations}">
+                    <f:option selected="${install.name == instance.ant.name}" value="${install.name}">
```
Comment (jglick) [inline src/main/resources/hudson/tasks/Ant/config.jelly]: BTW we ought to have a standard control in core for selecting a tool installation.

File: src/main/resources/hudson/tasks/Ant/config.jelly (hunk 8 above)
Comment (jglick) [reply src/main/resources/hudson/tasks/Ant/config.jelly]: (There are actually good reasons to discards the `if` block and show this dropdown unconditionally, but that would be out of scope for this PR.)

Comment (Wadeck) [reply src/main/resources/hudson/tasks/Ant/config.jelly]: > You tested this logic interactively, right?

I tested with and without installations but not sure to have cover all other potential cases there.
